patches/chromium/webview_fullscreen.patch

```diff
@@ -0,0 +1,35 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cheng Zhao <zcbenz@gmail.com>
+Date: Thu, 4 Oct 2018 14:57:02 -0700
+Subject: fix: also propagate fullscreen state for outer frame
+
+When entering fullscreen with Element.requestFullscreen in child frames,
+the parent frame should also enter fullscreen mode too. Chromium handles
+this for iframes, but not for webviews as they are essentially main
+frames instead of child frames.
+
+This patch makes webviews propagate the fullscreen state to embedder.
+
+Note that we also need to manually update embedder's
+`api::WebContents::IsFullscreenForTabOrPending` value.
+
```

Comment:
Is this something we can potentially upstream?

Reply:
The patch itself is not enough to fix the issue, so we can not upstream it directly.

To fix the problem we also have to make the fullscreen state sync between the webview and its embedder frame, and in Electron we are just manually synchronizing the states. A decent patch should change callers or callees of `IsFullscreenForTabOrPending` to be aware of outer frames (webviews), but it would require much more refactoring.

So I think it does not worth such efforts upstreaming the changes to Chromium, as very few people would encounter this problem in Chrome browser and the patch in this PR is very easy to maintain.